File: ValentinaJimenez_GenerarTiempos.cpp
#include <iostream>

using std::cout;
using std::endl;


// declarar funciones 

int fibonacci(int N);

float get_time(int N);

//la funcion recursiva es la que se llama a si misma, se utilizan factoriales para hallar factoriales
//

int fibonacci(int N) {

	//primero se define el caso base en el que la funcion es igual a n

	if( N==0 || N==1){

		return N;
	}
	// aqui ya es el recursivo, la funcion es igual a la funcion en n-1 y en n-2

	else{
	// declara la variable donde se va a guardar la funcion fibonacci
		int funcion;

		funcion=fibonacci(N-1)+fibonacci(N-2);
		return funcion;

	}
		
}

float get_time(int N){

	clock_t t;
	t = clock();
	fibonacci(N);
	// saca el tiempo que se demora en hacer la funcion fibonacci
	t = clock() - t;
	float time = ((float)t)/CLOCKS_PER_SEC;

}



//aqui se ejecuta las funciones declaradas al inicio 

int main(){

	int n=20;

	cout <<"La serie de Fibonacci es:"<< fibonacci(n) << endl;

	int N=0;
	while(N<=35){

	
	cout<< N <<","<< get_time(n) << endl;
	N++;

	}
	return 0;

}

	

	

File: ValentinaJimenez_GenerarTiempos.py
import numpy as np

# funcion fibonacci

def fibonacci(N):

	# condicion del caso base

	if(N==0 or N==1):

		return N

	# esto es para el caso recursivo

	else:

		funcion=fibonacci(N-1)+fibonacci(N-2)
		return funcion

# funcion que da el tiempo que tarda la funcion fibonacci para un entero N
def get_time(N):

	import time
	t0 = time.time()
	fibonacci(N)
	t1 = time.time()-t0
	return t1

# el while es necesario para que no lo haga infinitas veces, de esta manera lo tenemos para los N menores a 25
N=0;
while(N<=25):

	print N, ",", get_time(N)
	N+=1



